Kudu.Core/Jobs/TriggeredJobsManager.cs

```diff
@@ -31,7 +32,18 @@ public TriggeredJobsManager(ITraceFactory traceFactory, IEnvironment environment
 
         public override IEnumerable<TriggeredJob> ListJobs()
         {
-            return ListJobsInternal();
+            var cache = TriggeredJobCache;
+            if (cache == null)
+            {
+                cache = ListJobsInternal();
+                TriggeredJobCache = cache;
+            }
+            return cache;
+        }
```

Comment:
the watchers for both continuous and triggered jobs are not symmetric in implementation, for continuous it is in the same class but for the triggered it is in the scheduler. that led to the necessity of having the cache object as an internal static so that the scheduler watcher can null it when needed, such a thing can not work if it was pulled in a common class, so I think this should stay as is.